diff -Nru --exclude=.svn qPloneResolveUID.orig/transforms/ruid_to_url.py qPloneResolveUID/transforms/ruid_to_url.py
--- qPloneResolveUID.orig/transforms/ruid_to_url.py	2007-01-05 19:10:18.000000000 +0300
+++ qPloneResolveUID/transforms/ruid_to_url.py	2007-01-06 22:58:23.000000000 +0300
@@ -44,7 +44,7 @@
         for uid in unique_ruid:
             obj = rc.lookupObject(uid)
             if obj:
-                ruid_url[uid] = pu.getRelativeUrl(obj)
+                ruid_url[uid] = pu.getPortalPath() + '/' + pu.getRelativeUrl(obj)
         return ruid_url           
     
     def convert(self, orig, data, **kwargs):
@@ -63,4 +63,4 @@
 
 def register():
     return ruid_to_url()
-    
\ No newline at end of file
+    
